CommonUI/Sources/CommonUI/AsyncImage/AsyncImageViewBuilder.swift

```diff
@@ -0,0 +1,32 @@
+import SwiftUI
+
+public struct AsyncImageViewBuilder<Content: View>: View {
+    let url: URL
+    let content: (AsyncImagePhase) -> Content
+    var modifiers: (Image) -> Image = { $0 }
+
+    public init?(urlString: String, @ViewBuilder content: @escaping (AsyncImagePhase) -> Content) {
+        guard let movieURL = URL(string: urlString) else {
+            return nil
+        }
+        self.url = movieURL
+        self.content = content
+    }
+
+    public var body: some View {
+        
+        VStack {
+            AsyncImage(url: url, content: content)
+        }
+        .onAppear {
+            // Load and cache the image using URLSession
+            let request = URLRequest(url: url, cachePolicy: .returnCacheDataElseLoad)
+            URLSession.shared.dataTask(with: request) { data, response, _ in
+                if let data = data, let response = response {
+                    let cachedResponse = CachedURLResponse(response: response, data: data)
+                    URLCache.shared.storeCachedResponse(cachedResponse, for: request)
```

Comment:
Still not sure how exactly it works, but I tested it by turning on the off internet, it seems the Cache is working perfect.

Reply:
Since we are adding a URLCache policy of . returnCacheDataElseLoad and it is in the URLRequest, we'll need to apply this policy and override the default AsyncImage Caching.  We want to make sure by storing the data ourselves with the policy we indicated.  I would think we shouldn't have to call 'URLCache.shared.storeCachedResponse' but we'll need to make sure the cache policy is in effect.   I'll do more research on this, but Andrew testing proves that the cache is being used.